provider/github/watcher_test.go

```diff
@@ -108,11 +117,14 @@ func (s *WatcherTestSuite) TestWatch() {
 		return nil
 	})
 
+	mutex.Lock()
```

Comment:
do we really need this lock?
we check `s.EqualError(err, "context deadline exceeded")` it means `Watch` function stopped already and there shouldn't be any changes to this variables.

Reply:
Without it `go test -race` complains. I think that even if we end the `Watch` goroutine it might happen just when an http is in process, and the handler func would still be finish running and write callsA/B, even if the response goes nowhere.